### ISSUE TITLE
Potential issue with PhotonOS Docker host

### ISSUE THREAD

**Issue by si-heger** [NetApp/trident#248]
Following customer problem: Docker host (based on a VMware VM with PhotonOS 3.0 BUILD 26156e2 as host OS and Docker Version 18.06.2, build 6d37f41) and Trident 19.01 with ONTAP 9.3. They are able to create and delete volumes perfectly fine (docker create volume ...) but they can't mount it into a container. We are just using NFSv3 (v4 is disabled in ONTAP) with Export Policy 0.0.0.0/0. They can manually mount the volume on the docker host and they can ping the ONTAP LIFs from within the container.

Following error message occurs: 'May 02 08:15:45 dockerprivdmz1 dockerd[321]: time="2019-05-02T08:15:45+02:00" level=error msg="time=\"2019-05-02T06:15:45Z\" level=error msg=\"Docker frontend method returning error: error attaching volume nextcloud_mariadb, mountpoint /var/lib/docker-volumes/netapp/docker_nextcloud_mariadb, error: error mounting NFS volume 10.2.96.7:/docker_nextcloud_mariadb on mountpoint /var/lib/docker-volumes/netapp/docker_nextcloud_mariadb: exit status 32\"" plugin=f4687f6c0c17c6523e9e101a50996b73de70e3abd74fb55b8caf1e049278c1f1'

Here is the config.json
{
    "version": 1,
    "storageDriverName": "ontap-nas",
    "storagePrefix": "docker_",
    "managementLIF": "ltt.fullcustomerdomain.de",
    "dataLIF": "10.2.96.7",
    "svm": "ltt-svm",
    "username": "trident_user",
    "password": "PW",
    "aggregate": "ltt_data",
    "defaults": {
      "size": "10G",
      "spaceReserve": "none",
      "exportPolicy": "dockerprivdmz",
      "snapshotPolicy": "none",
      "snapshotReserve": "0"
    }
}

See details in screenshot
![docker1](https://user-images.githubusercontent.com/35133923/57081130-6d1dcb80-6cf4-11e9-8a5d-e678274ec8fb.png)
![docker2](https://user-images.githubusercontent.com/35133923/57081136-70b15280-6cf4-11e9-8df3-27757263f022.png)

**Comment by adkerr** [comment]
Can you try adding `"nfsMountOptions": "vers=3"` to your backend configuration and try again?

**Comment by si-heger** [comment]
Will do!

**Comment by si-heger** [comment]
Ok, we found the issue. Following a hint from jacoba in Slack and following Blog article https://www.exospheredata.com/2017/05/25/problem-attaching-docker-volume/ the fix is following:

Trident needs the rpcbind service running in Linux to work correctly. It seems it is enough to just start it with
`
systemctl enable rpcbind.service
systemctl start rpcbind.service
`

My proposal: Close this ticket as fixed but please include in the Trident documentation in "Troubleshotting" the fact that the rpcbind service needs to run (and maybe the connection to this strange error code 32)

**Comment by kremers** [comment]
Ensuring rpcbind is running can be done via daemon set

**Comment by scaleoutsean** [comment]
If you prepare the workers as per instructions (https://netapp-trident.readthedocs.io/en/stable-v19.07/kubernetes/operations/tasks/worker.html#worker-preparation), rpcbind is automatically installed as a dependency, at least on Ubuntu. Photon by default uses a different package manager, but the same packages that get installed by rpm should be installed on other Linux variants for which detailed steps aren't provided.

```
$ sudo apt-get install nfs-common
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libnfsidmap2 libtirpc1 rpcbind
Suggested packages:
  open-iscsi watchdog
The following NEW packages will be installed:
  libnfsidmap2 libtirpc1 nfs-common rpcbind
```

**Comment by gnarl** [comment]
The automated Node Prep is available as a beta feature in the Trident 20.10.0 release.
